### ISSUE TITLE
当我在 django 部署此项目时！

### ISSUE THREAD

**Issue by wangchuanlai** [myhub/tr#43]
心血来潮 部署到django 中，并发请求是抛出一下异常！
terminate called after throwing an instance of 'c10::Error'
  what():  storage() called on undefined Tensor (storage at /pytorch/c10/core/UndefinedTensorImpl.cpp:32)
frame #0: c10::Error::Error(c10::SourceLocation, std::string const&) + 0x33 (0x7fecc1712813 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libc10.so)
frame #1: c10::UndefinedTensorImpl::storage() const + 0xb2 (0x7fecc170c772 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libc10.so)
frame #2: <unknown function> + 0x2ab8528 (0x7fecc43ef528 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #3: std::tuple<at::Tensor, at::Tensor, at::Tensor> at::ATenOpTable::callUnboxed<std::tuple<at::Tensor, at::Tensor, at::Tensor>, at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, bool, double, double>(at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, bool, double, double) const + 0xca (0x7fecc25ce3ca in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #4: at::native::_batch_norm_impl_index(at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, bool, double, double, bool) + 0x242 (0x7fecc25ca662 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #5: <unknown function> + 0x10b81fc (0x7fecc29ef1fc in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #6: <unknown function> + 0x2d2d7ae (0x7fecc46647ae in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #7: at::native::batch_norm(at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, bool, double, double, bool) + 0x105 (0x7fecc25c8eb5 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #8: <unknown function> + 0x10b7edc (0x7fecc29eeedc in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #9: <unknown function> + 0x2d22449 (0x7fecc4659449 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #10: torch::nn::BatchNormImpl::pure_forward(at::Tensor const&, at::Tensor const&, at::Tensor const&) + 0x183 (0x7fecc4be12a3 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #11: torch::nn::BatchNormImpl::forward(at::Tensor const&) + 0x27 (0x7fecc4be14f7 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #12: <unknown function> + 0x8da99 (0x7fecce896a99 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/libtr.so)
frame #13: tr_recognize + 0x171 (0x7fecce87e3a1 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/libtr.so)
frame #14: ffi_call_unix64 + 0x4c (0x7fece9cd8e20 in /usr/lib/python3.5/lib-dynload/_ctypes.cpython-35m-x86_64-linux-gnu.so)
frame #15: ffi_call + 0x2eb (0x7fece9cd888b in /usr/lib/python3.5/lib-dynload/_ctypes.cpython-35m-x86_64-linux-gnu.so)
frame #16: _ctypes_callproc + 0x49a (0x7fece9cd301a in /usr/lib/python3.5/lib-dynload/_ctypes.cpython-35m-x86_64-linux-gnu.so)
frame #17: <unknown function> + 0x9fcb (0x7fece9cc6fcb in /usr/lib/python3.5/lib-dynload/_ctypes.cpython-35m-x86_64-linux-gnu.so)
frame #18: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #19: PyEval_EvalFrameEx + 0x4f2f (0x5354af in /root/pythonVen/bin/python)
frame #20: PyEval_EvalFrameEx + 0x4b64 (0x5350e4 in /root/pythonVen/bin/python)
frame #21: /root/pythonVen/bin/python() [0x539a13]
frame #22: PyEval_EvalFrameEx + 0x5122 (0x5356a2 in /root/pythonVen/bin/python)
frame #23: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #24: /root/pythonVen/bin/python() [0x4e3537]
frame #25: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #26: PyEval_EvalFrameEx + 0x24a2 (0x532a22 in /root/pythonVen/bin/python)
frame #27: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #28: /root/pythonVen/bin/python() [0x4e3537]
frame #29: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #30: PyEval_EvalFrameEx + 0x24a2 (0x532a22 in /root/pythonVen/bin/python)
frame #31: PyEval_EvalCodeEx + 0x88a (0x53af6a in /root/pythonVen/bin/python)
frame #32: /root/pythonVen/bin/python() [0x4e3537]
frame #33: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #34: PyEval_EvalFrameEx + 0x24a2 (0x532a22 in /root/pythonVen/bin/python)
frame #35: PyEval_EvalFrameEx + 0x4b64 (0x5350e4 in /root/pythonVen/bin/python)
frame #36: /root/pythonVen/bin/python() [0x539f5f]
frame #37: PyEval_EvalFrameEx + 0x5572 (0x535af2 in /root/pythonVen/bin/python)
frame #38: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #39: /root/pythonVen/bin/python() [0x4e3423]
frame #40: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #41: /root/pythonVen/bin/python() [0x4f08be]
frame #42: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #43: /root/pythonVen/bin/python() [0x55fbf6]
frame #44: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #45: PyEval_EvalFrameEx + 0x4f2f (0x5354af in /root/pythonVen/bin/python)
frame #46: /root/pythonVen/bin/python() [0x539f5f]
frame #47: PyEval_EvalFrameEx + 0x5572 (0x535af2 in /root/pythonVen/bin/python)
frame #48: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #49: /root/pythonVen/bin/python() [0x4e3423]
frame #50: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #51: /root/pythonVen/bin/python() [0x4f08be]
frame #52: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #53: /root/pythonVen/bin/python() [0x55fbf6]
frame #54: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #55: PyEval_EvalFrameEx + 0x4f2f (0x5354af in /root/pythonVen/bin/python)
frame #56: /root/pythonVen/bin/python() [0x539f5f]
frame #57: PyEval_EvalFrameEx + 0x5572 (0x535af2 in /root/pythonVen/bin/python)
frame #58: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #59: /root/pythonVen/bin/python() [0x4e3423]
frame #60: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #61: /root/pythonVen/bin/python() [0x4f08be]
frame #62: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #63: /root/pythonVen/bin/python() [0x55fbf6]

(pythonVen) root@123321:~/chengyu# python manage.py runserver 0.0.0.0:89
Watching for file changes with StatReloader
Performing system checks...

# tr 1.5.0 https://github.com/myhub/tr
System check identified no issues (0 silenced).

You have 17 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): admin, auth, contenttypes, sessions.
Run 'python manage.py migrate' to apply them.

March 30, 2020 - 17:41:25
Django version 2.2.5, using settings 'chengyu.settings'
Starting development server at http://0.0.0.0:89/
Quit the server with CONTROL-C.
terminate called after throwing an instance of 'c10::Error'
  what():  Expected a Tensor of type Variable but found an undefined Tensor for argument #1 'other' (checked_cast_variable at /pytorch/torch/csrc/autograd/VariableTypeManual.cpp:37)
frame #0: c10::Error::Error(c10::SourceLocation, std::string const&) + 0x33 (0x7f5061fde813 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libc10.so)
frame #1: <unknown function> + 0x3246630 (0x7f5065449630 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #2: <unknown function> + 0x3246889 (0x7f5065449889 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #3: <unknown function> + 0x2c9cf6b (0x7f5064e9ff6b in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/../libtorch/lib/libtorch.so)
frame #4: <unknown function> + 0x869ba (0x7f506f15b9ba in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/libtr.so)
frame #5: at::Tensor::add_(at::Tensor const&, c10::Scalar) const + 0x10b (0x7f506f15bccb in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/libtr.so)
frame #6: <unknown function> + 0x8d3b2 (0x7f506f1623b2 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/libtr.so)
frame #7: tr_recognize + 0x171 (0x7f506f14a3a1 in /root/pythonVen/lib/python3.5/site-packages/tr-1.5.0-py3.5.egg/tr/libtr.so)
frame #8: ffi_call_unix64 + 0x4c (0x7f508a5cae20 in /usr/lib/python3.5/lib-dynload/_ctypes.cpython-35m-x86_64-linux-gnu.so)
frame #9: ffi_call + 0x2eb (0x7f508a5ca88b in /usr/lib/python3.5/lib-dynload/_ctypes.cpython-35m-x86_64-linux-gnu.so)
frame #10: _ctypes_callproc + 0x49a (0x7f508a5c501a in /usr/lib/python3.5/lib-dynload/_ctypes.cpython-35m-x86_64-linux-gnu.so)
frame #11: <unknown function> + 0x9fcb (0x7f508a5b8fcb in /usr/lib/python3.5/lib-dynload/_ctypes.cpython-35m-x86_64-linux-gnu.so)
frame #12: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #13: PyEval_EvalFrameEx + 0x4f2f (0x5354af in /root/pythonVen/bin/python)
frame #14: PyEval_EvalFrameEx + 0x4b64 (0x5350e4 in /root/pythonVen/bin/python)
frame #15: /root/pythonVen/bin/python() [0x539a13]
frame #16: PyEval_EvalFrameEx + 0x5122 (0x5356a2 in /root/pythonVen/bin/python)
frame #17: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #18: /root/pythonVen/bin/python() [0x4e3537]
frame #19: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #20: PyEval_EvalFrameEx + 0x24a2 (0x532a22 in /root/pythonVen/bin/python)
frame #21: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #22: /root/pythonVen/bin/python() [0x4e3537]
frame #23: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #24: PyEval_EvalFrameEx + 0x24a2 (0x532a22 in /root/pythonVen/bin/python)
frame #25: PyEval_EvalCodeEx + 0x88a (0x53af6a in /root/pythonVen/bin/python)
frame #26: /root/pythonVen/bin/python() [0x4e3537]
frame #27: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #28: PyEval_EvalFrameEx + 0x24a2 (0x532a22 in /root/pythonVen/bin/python)
frame #29: PyEval_EvalFrameEx + 0x4b64 (0x5350e4 in /root/pythonVen/bin/python)
frame #30: /root/pythonVen/bin/python() [0x539f5f]
frame #31: PyEval_EvalFrameEx + 0x5572 (0x535af2 in /root/pythonVen/bin/python)
frame #32: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #33: /root/pythonVen/bin/python() [0x4e3423]
frame #34: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #35: /root/pythonVen/bin/python() [0x4f08be]
frame #36: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #37: /root/pythonVen/bin/python() [0x55fbf6]
frame #38: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #39: PyEval_EvalFrameEx + 0x4f2f (0x5354af in /root/pythonVen/bin/python)
frame #40: /root/pythonVen/bin/python() [0x539f5f]
frame #41: PyEval_EvalFrameEx + 0x5572 (0x535af2 in /root/pythonVen/bin/python)
frame #42: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #43: /root/pythonVen/bin/python() [0x4e3423]
frame #44: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #45: /root/pythonVen/bin/python() [0x4f08be]
frame #46: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #47: /root/pythonVen/bin/python() [0x55fbf6]
frame #48: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #49: PyEval_EvalFrameEx + 0x4f2f (0x5354af in /root/pythonVen/bin/python)
frame #50: /root/pythonVen/bin/python() [0x539f5f]
frame #51: PyEval_EvalFrameEx + 0x5572 (0x535af2 in /root/pythonVen/bin/python)
frame #52: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #53: /root/pythonVen/bin/python() [0x4e3423]
frame #54: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #55: /root/pythonVen/bin/python() [0x4f08be]
frame #56: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #57: /root/pythonVen/bin/python() [0x55fbf6]
frame #58: PyObject_Call + 0x47 (0x5c3bd7 in /root/pythonVen/bin/python)
frame #59: PyEval_EvalFrameEx + 0x4f2f (0x5354af in /root/pythonVen/bin/python)
frame #60: /root/pythonVen/bin/python() [0x539f5f]
frame #61: PyEval_EvalFrameEx + 0x5572 (0x535af2 in /root/pythonVen/bin/python)
frame #62: PyEval_EvalCodeEx + 0x13b (0x53a81b in /root/pythonVen/bin/python)
frame #63: /root/pythonVen/bin/python() [0x4e3423]

**Comment by myhub** [comment]
不支持并发请求的

**Comment by myhub** [comment]
tr 2.0 is updated, more faster, more simple
